doc/usage.rst

```diff
@@ -204,21 +207,34 @@ Then, a :any:`Driver` needs to be created on the `Target`::
   >>> sd = SerialDriver(t, name=None)
 
 
-As the `SerialDriver` declares a binding to a SerialPort, the target binds it
+As the :any:`SerialDriver` declares a binding to a :any:`SerialPort`, the target binds it
 to the resource created above::
 
   >>> sd.port
   RawSerialPort(target=Target(name='example', env=None), name=None, state=<BindingState.bound: 1>, avail=True, port='/dev/ttyUSB0', speed=115200)
   >>> sd.port is rsp
   True
 
-Before the driver can be used, it needs to be activated::
+Driver Activation
+^^^^^^^^^^^^^^^^^
+Before the driver can be used, it needs to be activated.
+On driver activation, labgrid makes sure that all resources needed by the
+driver are available::
 
   >>> t.activate(sd)
   >>> sd.write(b'test')
 
-Active drivers can be accessed by class (any `Driver` or `Protocol`) using some
-syntactic sugar::
+If a resource is not available (or not available after a certain timeout,
```

Comment:
Hi, am also trying to figure out the concepts and terminology behind labgrid. :)

As far as I understood, the word "resource" perhaps isn't the best choice of words for this context.
It seems there are two things "resources" might try to address:

* the ["Resource"](https://labgrid.readthedocs.io/en/latest/overview.html#resources) as introduced by labgrid.
* the "actual resource" to which the above is a proxy. I.e. `/dev/ttyUSB0`

From what I've been seeing it's not possible to even create a driver if the correct "Resource" isn't present on the target. This happens a step before activation and is a requirement of the binding phase. It would perhaps be better to use another term for the "actual resource" to describe what might happen in the activation phase.

In this example where I'm mostly trying to describe the implicit assumptions in the code I describe the "actual resource" as an "object below the driver which a driver consumes". It's not great wording either :) 

```
username = "pi"
password = "raspberry"                                                                                                                                                                    
                                                                                                                                                                                                         
rpi = Target("RPi")                                                                                                                                                                                                
                                                                                                                                                                                                           
network_service = NetworkService(rpi,                                                                                                                                                                      
                                 address="192.168.18.210",                                                                                                                                                 
                                 name="rpi_network",                                                                                                                                                       
                                 username=username,                                                                                                                             
                                 password=password)
raw_serial_port = RawSerialPort(rpi, None, port="/dev/ttyUSB0", speed=115200)                                                                                                                                      
# At this point both resources belong to the target. This happened implicitly.                                                                                                                                     
 
                                                                                                                                                                                                                   
ssh_driver = SSHDriver(rpi, name=None)                                                                                                                                                                             
serial_driver = SerialDriver(rpi, name=None)                                                                                                                                                                       
prompt_regex_pattern = f'{username}' + r'@\w+:[^ ]+ '                                                                                                                                                              
login_regex_pattern = ' login: '
shell_driver = ShellDriver(rpi, name=None,                                                                                                                                                                         
                           prompt=prompt_regex_pattern,                                                                                                                                                            
                           login_prompt=login_regex_pattern,                                                                                                                                                       
                           username=username,                                                                                                                                                                      
                           password=password)                                                                                                                                                                      
                                                                                                                                                                                                                   
# By belonging to the same target, the drivers automatically bounded to the resources and other drivers that fit their requirements.                                                                               
# The SSHDriver bounded to NetworkService
# The ShellDriver bounded to SerialDriver whicb bounded to RawSerialPort                                                                                                                                  
                                                                                                                                                                                                                   
rpi.activate(ssh_driver)
rpi.activate(shell_driver)                                                                                                                                                                                         
                                                                                                                                                                                                                   
# When activated the drivers consume the object below it (either the resource or the other driver).                                                                                                                
# After that they are ready to be used
```

Reply:
We could use "hardware resource" and "Resource object", would that be better?

Reply:
Don't have a clear one for that. :/
While writing the comment above a `resource on the level of the OS` seemed like a good candidate.

Reply:
OK, I think I've got all of this in the current version now. Sometimes I just used "hardware" when it is clear what is meant.